packages/driver/cypress/integration/e2e/multi-domain/snapshots/multi_domain_snapshot_location.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { findCrossOriginLogs } from '../../../../support/utils'
+
+// @ts-ignore / session support is needed for visiting about:blank between tests
+context('cross-origin snapshot location', { experimentalSessionSupport: true }, () => {
+  let logs: Map<string, any>
+
+  beforeEach(() => {
+    logs = new Map()
+
+    cy.on('log:changed', (attrs, log) => {
+      logs.set(attrs.id, log)
+    })
+
+    cy.visit('/fixtures/multi-domain.html')
+    cy.get('a[data-cy="cross-origin-secondary-link"]').click()
+  })
+
+  it('.hash()', (done) => {
+    cy.on('command:queue:end', () => {
+      setTimeout(() => {
+        const { crossOriginLog, consoleProps } = findCrossOriginLogs('hash', logs, 'foobar.com')
+
+        expect(crossOriginLog).to.be.true
+        expect(consoleProps.Command).to.equal('hash')
+
+        done()
+      }, 250)
+    })
+
+    cy.origin('http://foobar.com:3500', () => {
+      cy.hash()
+    })
+  })
+
+  it('.location()', (done) => {
+    cy.on('command:queue:end', () => {
+      setTimeout(() => {
```

Comment:
I still need to find a better method here to complete these tests. Going to explore this soon and am open to suggestions!